drivers/display/display_mcux_elcdif.c

```diff
@@ -19,34 +20,28 @@
 
 LOG_MODULE_REGISTER(display_mcux_elcdif, CONFIG_DISPLAY_LOG_LEVEL);
 
-/* Pixel formats */
-#define MCUX_ELCDIF_PIXEL_FORMAT_RGB888 0U
-#define MCUX_ELCDIF_PIXEL_FORMAT_BGR565 1U
-
-K_HEAP_DEFINE(mcux_elcdif_pool,
-	      CONFIG_MCUX_ELCDIF_POOL_BLOCK_MAX *
-	      CONFIG_MCUX_ELCDIF_POOL_BLOCK_NUM);
+#ifdef CONFIG_MCUX_ELCDIF_DOUBLE_FRAMEBUFFER
+#define MCUX_ELCDIF_FB_NUM 2
+#else
+#define MCUX_ELCDIF_FB_NUM 1
+#endif
 
 struct mcux_elcdif_config {
 	LCDIF_Type *base;
 	void (*irq_config_func)(const struct device *dev);
 	elcdif_rgb_mode_config_t rgb_mode;
-	uint8_t pixel_format;
+	enum display_pixel_format pixel_format;
+	size_t pixel_bytes;
+	size_t fb_bytes;
```

Comment:
Can we leave these inside `struct mcux_elcdif_data`. This would minimize the changes needed.

Reply:
I tend to disagree, all static options ideally go into the device's `config`, this way we save precious bytes in RAM.